src/api/app/controllers/webui/cloud/upload_jobs_controller.rb

```diff
@@ -0,0 +1,72 @@
+module Webui
+  module Cloud
+    class UploadJobsController < WebuiController
+      before_action :require_login
+      before_action -> { feature_active?(:cloud_upload) }
+      before_action :set_user
+      before_action :validate_configuration_presence
+      before_action :set_breadcrump
+
+      def index
+        @upload_jobs = @user.upload_jobs
+      end
+
+      def new
+        @project_name = params[:project]
+        @package_name = params[:package]
+        @repository_name = params[:repository]
+        @architecture_name = params[:architecture]
+        @file_name = params[:filename]
+      end
+
+      def create
+        begin
+          result = Backend::Api::Cloud.upload(
+            permitted_params[:project],
+            permitted_params[:package],
+            permitted_params[:repository],
+            permitted_params[:architecture],
+            permitted_params[:filename],
+            @user,
+            permitted_params[:region]
+          )
```

Comment:
Basically you deconstruct the hash here and recreate it [later](https://github.com/openSUSE/open-build-service/pull/4260/files#diff-cd1d47a3124eb8adfbb326638f938d5cR9). I would instead hand over a hash here.

Reply:
Yeah you're right!